spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -165,25 +165,32 @@ public static String sourceName(String target, String applicationName, String na
 	 * defined order).
 	 */
 	public static MultipleSourcesContainer processNamedData(List<StrippedSourceContainer> strippedSources,
-			Environment environment, Set<String> sourceNames, String namespace, boolean decode) {
+			Environment environment, LinkedHashSet<String> sourceNames, String namespace, boolean decode) {
 
-		Set<String> foundSourceNames = new HashSet<>();
+		Map<String, StrippedSourceContainer> hashByName = strippedSources.stream()
+				.collect(Collectors.toMap(StrippedSourceContainer::name, Function.identity()));
+
+		LinkedHashSet<String> foundSourceNames = new LinkedHashSet<>();
 		Map<String, Object> data = new HashMap<>();
 
-		strippedSources.stream().filter(source -> sourceNames.contains(source.name())).collect(Collectors.toList())
-				.forEach(foundSource -> {
-					String sourceName = foundSource.name();
-					LOG.debug("Found source with name : '" + sourceName + " in namespace: '" + namespace + "'");
-					foundSourceNames.add(sourceName);
-					// see if data is a single yaml/properties file and if it needs
-					// decoding
-					Map<String, String> rawData = foundSource.data();
-					if (decode) {
-						rawData = decodeData(rawData);
-					}
-					data.putAll(SourceDataEntriesProcessor.processAllEntries(rawData == null ? Map.of() : rawData,
-							environment));
-				});
+		// this is an ordered stream, and it means that non-profile based sources will be
```

Comment:
this is the idea of this PR. In plain Spring, every if we have `application.yaml` and `application-dev.yaml`, properties from the latter will override the ones from the former. I think that users using our project would expect the same: `configmap-dev` will override properties from `configmap`. Me, personally, would expect that. 

This PR enforces that order, also adds tests to make sure this is respected.